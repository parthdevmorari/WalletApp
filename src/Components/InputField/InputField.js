import React from 'react';
import {View, Text} from 'react-native';
import styles from './InputFieldStyles';
import {TextInput} from '@Component';

import Icon from 'react-native-vector-icons/Entypo';
import Icon2 from 'react-native-vector-icons/Ionicons';

export const InputField = (props) => {
  const [isVisible, setIsVisible] = React.useState(false);
  const [icon, setIcon] = React.useState('eye-with-line');

  const {
    title,
    mainContainerStyle,
    multiline,
    inputContainerStyle,
    titleStyle,
    value,
    iconStyle,
    textFieldStyle,
    placeholder,
    onChangeText,
    keyboardType,
    readOnly,
    onFocus,
    onRef,
    secureTextEntry,
    error,
    onBlur,
    disable,
    length,
    iconContainerStyle
  } = props;
  React.useEffect(() => {
    if (secureTextEntry) {
      setIsVisible(true);
    }
  }, [secureTextEntry]);
  const _changeIcon = () => {
    setIcon(icon === 'eye' ? 'eye-with-line' : 'eye');
    setIsVisible(!isVisible);
  };
  return (
    <View style={[styles.inputBoxContainer, mainContainerStyle]}>
      {title && (
        <Text style={[styles.inputLableTaxt, titleStyle]}>{title}</Text>
      )}
      <View style={[styles.InputRow, inputContainerStyle]}>
	  {props.leftIcon && (
          <View style={[styles.leftIcon, iconContainerStyle]}>
            <Icon size={20} name={props.leftIcon} onPress={() => _changeIcon()}  style={[styles.icon,iconStyle]}/>
          </View>
        )}
         {props.leftIcon2 && (
          <View style={[styles.leftIcon, iconContainerStyle]}>
            <Icon2 size={20} name={props.leftIcon2} onPress={() => _changeIcon()}  style={[styles.icon ,iconStyle]}/>
          </View>
        )}
        <View
          style={{
            width:( props.secureTextEntry )? '90%' : '100%',
            justifyContent: 'center',
            alignItems: 'center',
          }}>
			  
          <TextInput
            textFieldStyle={[styles.inputContainer, textFieldStyle]}
            placeholder={placeholder}
            value={value}
            onChangeText={(text) => onChangeText(text)}
            onFocus={() => {
              if (onFocus !== undefined) {
                onFocus();
              }
            }}
            onBlur={() => {
              if (onBlur !== undefined) {
                onBlur();
              }
            }}
            length={length}
            error={error}
            secureTextEntry={isVisible ? true : false}
            onRef={onRef}
            editable={readOnly ? false : true}
            keyboardType={keyboardType}
            disable={disable}
            multiline={multiline}
          />
        </View>

        {props.secureTextEntry && props.isRight && (
          <View style={[styles.iconRowpass, iconStyle]}>
            <Icon size={20} name={icon} onPress={() => _changeIcon()} />
          </View>
        )}
      </View>
    </View>
  );
};

export default InputField;
