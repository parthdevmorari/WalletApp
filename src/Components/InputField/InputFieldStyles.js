import { StyleSheet, Dimensions, StatusBar } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { color } from '../../Constants/Color';
import { theme } from "../../Constants/Theme";

const width = Dimensions.get('window').width;
const styles = StyleSheet.create({
	inputBoxContainer: {
		marginBottom:wp(2),
	},
	inputLableTaxt: {
		fontSize:wp(4),
		fontWeight:'700',
		marginBottom:wp(2),
		color:theme.LABLE_COLOR,
	},
	InputRow: {
		borderWidth:1,
		borderColor:theme.PRIMARY_BORDER_COLOR,
		// height:wp(11),
		borderRadius:3,
		flexDirection:'row',
		backgroundColor:theme.action_Buttons_Text_Color,
	},
	inputContainer: {
		minHeight:wp(11),
		paddingHorizontal:10,
	}, 
	iconRowpass:{
        position: 'absolute',
		right: 3,
		top:3,
        height: 40,
        width: 35,
        padding: 5,
        flexDirection: 'row',
    },
	leftIcon:{
		justifyContent:'center',
    },
	icon:{
		textAlignVertical:'center',
		textAlign:'center',
	},
	

});
export default styles;

