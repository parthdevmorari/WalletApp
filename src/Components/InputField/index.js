import component from './InputField';
export default component;