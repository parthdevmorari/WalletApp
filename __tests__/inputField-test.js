import 'react-native';
import React from 'react';
import InputField from '../src/Components/InputField/InputField'
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

describe('InputField Component', () => {
  test('InputField Component', () => {
    const wrapper = shallow(<InputField />);
    expect(wrapper.exists()).toBe(true);
  });
  it('InputField  toMatchSnapshot', () => {
    const wrapper = shallow(<InputField />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});