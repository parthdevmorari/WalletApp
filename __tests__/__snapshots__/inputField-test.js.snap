// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField Component InputField  toMatchSnapshot 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 15,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderBottomWidth": 1,
          "borderColor": "#cbcbcb",
          "borderRadius": 3,
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <TextField
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        secureTextEntry={false}
        textFieldStyle={
          Array [
            Object {
              "minHeight": 82.5,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
